docs/getting-started/application/patterns.rst

```diff
@@ -53,122 +53,61 @@ The above policy controls access to the Expense model on the **record level**.
-data**.  It relies on other data to make the decision: the submitter of the
-expense (``expense.submitted_by``), and the manager of the submitter
-(``submitted_by.manager``).  This *other* data is called **secondary
-authorization data**.  An important class of secondary authorization data is
-**actor data**.  This data includes properties of the actor, like their role, or
-team membership that is often used in controlling access regardless of whether
-it is over rows, columns or fields.
-
-Where the Policy is Evaluated
------------------------------
-
-Where the policy is evaluated has a significant impact on the granularity of
-access control that is possible.  In the above example, we rely on the
-``amount`` field of the expense. Therefore, the ``Expense`` (the **primary
-authorization data**) must be fetched from the application's store when the rule
-is evaluated.
-
-.. _second-record-level:
-
-.. code-block:: python
-    :caption: :fab:`python`
-
-    def get_expense(user, expense_id):
-        expense = db.fetch(
-            "SELECT * FROM expenses WHERE id = %", expense_id)
-
-        if oso.is_allowed(user, "view", expense):
-            # Process request
-            ...
-        else:
-            # Not authorized
-            return NotAuthorizedResponse()
-
-This **policy evaluation point** is **after primary data fetch**. An
-authorization decision is made after the primary data is fetched from the
-persistence layer (be it a SQL database, an ORM, or another service) and can be
-used to make an authorization decisions.
-
-Alternatively, we could have placed our policy evaluation point **before primary
-data fetch**. This would limit the power of our policy, since we would not be
-able to check the ``amount`` field of the ``Expense``. Keep reading to see how
-we would apply this technique to **model level** and **field level** access
-control.
-
-.. tip::
-
-    A best practice for this type of access control is to integrate the policy
-    evaluation point within the **data access layer**.  This
-    ensures that the ``oso.is_allowed`` call is made when an expense is accessed, no
-    matter where that access occurs in the application.
-
-.. todo::
-
-   Would be nice to be able to say here: "see this guide for an
-   example of how to do this..."
-
 Model Level Access Control
 ==========================
 
-Sometimes, access control does not rely on properties of the primary data.  This
-type of access control is called **model level**.
+Sometimes, access control does not depend on the properties of the resource but just
+on it's type. This is called **model level** can be used for coarse level authorization.
+This is useful if you want to restrict access to a whole page if the user cannot see any
+of the resources of the kind the page is about. Since the rules depend only on the type
+of the model and not on the data it makes it easier to add these checks upstream in
+middleware.
```

Comment:
This sentence is a little vague. Is this referring to some kind of request middleware?